Installation issue: eckit@1.32 is not configuring all compression backends correctly

### Steps to reproduce the issue

eckit@1.32 added a new compression backend, but the spack package didn't contain the necessary updates to handle it: https://github.com/ecmwf/eckit/blame/develop/CMakeLists.txt


### Error message

```
==> No binary for eckit-1.32.3-lwwykhiayqmfqnifcrcrdzuufvg663yr found: installing from source
==> Warning: Intel's compilers may or may not optimize to the same degree for non-Intel microprocessors for optimizations that are not unique to Intel microprocessors
==> Using cached archive: /p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/cache/source_cache/_source-cache/archive/33/33e0fac2656cdd2f2d877dbfe7a4751ee657ab732c00dd90bd48a406298a100f.tar.gz
==> No patches needed for eckit
==> eckit: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    '/p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/envs/ne-oneapi-2024.2.0-build/install/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/gcc/12.2.0/cmake-3.31.8-4gtggwa/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/envs/ne-oneapi-2024.2.0-build/install/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/intel-oneapi-compilers/2024.2.0/eckit-1.32.3-lwwykhi' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/envs/ne-oneapi-2024.2.0-build/install/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/intel-oneapi-compilers/2024.2.0/eckit-1.32.3-lwwykhi/lib;/p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/envs/ne-oneapi-2024.2.0-build/install/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/intel-oneapi-compilers/2024.2.0/eckit-1.32.3-lwwykhi/lib64' '-DCMAKE_PREFIX_PATH:STRING=/p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/envs/ne-oneapi-2024.2.0-build/install/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/gcc/12.2.0/bison-3.8.2-yj5oc7m;/p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/envs/ne-oneapi-2024.2.0-build/install/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/intel-oneapi-compilers/2024.2.0/bzip2-1.0.8-2m6l4df;/p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/envs/ne-oneapi-2024.2.0-build/install/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/gcc/12.2.0/cmake-3.31.8-4gtggwa;/p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/envs/ne-oneapi-2024.2.0-build/install/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/none/none/compiler-wrapper-1.0-toutdiv;/p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/envs/ne-oneapi-2024.2.0-build/install/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/intel-oneapi-compilers/2024.2.0/ecbuild-3.12.0-2qvlmat;/p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/envs/ne-oneapi-2024.2.0-build/install/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/intel-oneapi-compilers/2024.2.0/eigen-3.4.0-yanbufc;/p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/envs/ne-oneapi-2024.2.0-build/install/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/intel-oneapi-compilers/2024.2.0/flex-2.6.4-25tziv2;/p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/envs/ne-oneapi-2024.2.0-build/install/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/gcc/12.2.0/gmake-4.4.1-k3cxlm5;/p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/envs/ne-oneapi-2024.2.0-build/install/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/intel-oneapi-compilers/2024.2.0/lz4-1.10.0-nxhpxiy;/p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/envs/ne-oneapi-2024.2.0-build/install/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/intel-oneapi-compilers/2024.2.0/ncurses-6.5-4om7pw5;/opt/cray/pe/mpich/8.1.26/ofi/intel/2022.1;/opt/intel/oneapi_2024.2.0.634;/opt/intel/oneapi_2024.2.0.634;/opt/intel/oneapi_2024.2.0.634' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_POLICY_DEFAULT_CMP0090:STRING=NEW' '-DCMAKE_FIND_USE_PACKAGE_REGISTRY:BOOL=OFF' '-DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON' '-DENABLE_EXPERIMENTAL:BOOL=OFF' '-DENABLE_BUILD_TOOLS:BOOL=ON' '-DENABLE_MPI:BOOL=ON' '-DENABLE_OMP:BOOL=ON' '-DENABLE_ECKIT_CMD:BOOL=ON' '-DENABLE_ECKIT_SQL:BOOL=ON' '-DENABLE_EIGEN:BOOL=ON' '-DENABLE_ARMADILLO:BOOL=OFF' '-DENABLE_MKL:BOOL=OFF' '-DENABLE_BZIP2:BOOL=ON' '-DENABLE_SNAPPY:BOOL=OFF' '-DENABLE_LZ4:BOOL=ON' '-DENABLE_AEC:BOOL=OFF' '-DENABLE_XXHASH:BOOL=ON' '-DENABLE_SSL:BOOL=OFF' '-DENABLE_CURL:BOOL=OFF' '-DENABLE_JEMALLOC:BOOL=OFF' '-DENABLE_UNICODE:BOOL=ON' '-DENABLE_AIO:BOOL=ON' '-DENABLE_TESTS:BOOL=OFF' '-DENABLE_EXTRA_TESTS:BOOL=OFF' '-DENABLE_DOCS:BOOL=OFF' '-DENABLE_CUDA:BOOL=OFF' '-DENABLE_VIENNACL:BOOL=OFF' '-DENABLE_RADOS:BOOL=OFF' '-DENABLE_RSYNC:BOOL=OFF' '-DENABLE_SANDBOX:BOOL=OFF' '-DBUILD_SHARED_LIBS=ON' '-DENABLE_LAPACK:BOOL=ON' '-DCURSES_NEED_NCURSES:BOOL=ON' '/p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/cache/build_stage/neptune/spack-stage-eckit-1.32.3-lwwykhiayqmfqnifcrcrdzuufvg663yr/spack-src'

1 error found in build log:
     91    -- eckit FOUND LZ4
     92    --    LZ4_INCLUDE_DIRS : [/p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/envs/ne-oneapi-2024.2.0-build/install/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_plac
           eholder__/__spack_path_placeholder__/intel-oneapi-compilers/2024.2.0/lz4-1.10.0-nxhpxiy/include]
     93    --    LZ4_LIBRARIES : [/p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/envs/ne-oneapi-2024.2.0-build/install/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeho
           lder__/__spack_path_placeholder__/intel-oneapi-compilers/2024.2.0/lz4-1.10.0-nxhpxiy/lib/liblz4.so]
     94    -- Found package LZ4 required for feature LZ4
     95    -- Feature LZ4 enabled
     96    -- Feature AEC disabled
  >> 97    CMake Error at /usr/lib64/cmake/libzip/libzip-targets.cmake:92 (message):
     98      The imported target "libzip::zipcmp" references the file
     99
     100        "/usr/bin/zipcmp"
     101
     102     but this file does not exist.  Possible reasons include:
     103

See build log for details:
  /p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/cache/build_stage/neptune/spack-stage-eckit-1.32.3-lwwykhiayqmfqnifcrcrdzuufvg663yr/spack-build-out.txt
```

### Information on your system

```
* **Spack:** 1.0.1 (https://github.com/spack/spack/commit/553caf208c8d5983bf86d6eb7c63fcb546095b7d)
* **Builtin repo:** /p/app/projects/NEPTUNE/spack-stack/spack-stack-v2-rc1/repos/builtin/repos/spack_repo/builtin (9e174ef)
* **Python:** 3.6.15
* **Platform:** linux-sles15-zen2
```


### Additional information

I am one of the package maintainers and I am going to fix this today, therefore not uploading any log files or notifying any other maintainers just yet.

I may ask for help to come up with a more elegant solution/syntax than I was able to.

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/eckit/package.py
+++ b/repos/spack_repo/builtin/packages/eckit/package.py
@@ -56,10 +56,20 @@ class Eckit(CMakePackage):
         values=any_combination_of("eigen", "armadillo", "mkl", "lapack"),
         description="List of supported linear algebra backends",
     )
+
+    # There is probably a more elegant way to handle the differences
+    # in valid compression backends, but this works ...
     variant(
         "compression",
         values=any_combination_of("bzip2", "snappy", "lz4", "aec"),
         description="List of supported compression backends",
+        when="@:1.31",
+    )
+    variant(
+        "compression",
+        values=any_combination_of("bzip2", "snappy", "lz4", "aec", "zip"),
+        description="List of supported compression backends",
+        when="@1.32:",
     )
     variant("xxhash", default=True, description="Enable xxHash support for hashing")
     variant("ssl", default=False, description="Enable MD4 and SHA1 support with OpenSSL")
@@ -101,6 +111,7 @@ class Eckit(CMakePackage):
     depends_on("snappy", when="compression=snappy")
     depends_on("lz4", when="compression=lz4")
     depends_on("libaec", when="compression=aec")
+    depends_on("libzip", when="compression=zip")
 
     depends_on("openssl", when="+ssl")
 
@@ -144,6 +155,7 @@ class Eckit(CMakePackage):
             self.define("ENABLE_SNAPPY", "compression=snappy" in self.spec),
             self.define("ENABLE_LZ4", "compression=lz4" in self.spec),
             self.define("ENABLE_AEC", "compression=aec" in self.spec),
+            self.define("ENABLE_ZIP", "compression=zip" in self.spec),
             self.define_from_variant("ENABLE_XXHASH", "xxhash"),
             self.define_from_variant("ENABLE_SSL", "ssl"),
             self.define_from_variant("ENABLE_CURL", "curl"),

